File: src/main/java/com/workforce/hrm/service/UserService.java
package com.workforce.hrm.service;

import com.workforce.hrm.dto.request.ChangePasswordRequest;
import com.workforce.hrm.dto.request.ResetPasswordRequest;
import com.workforce.hrm.dto.request.UpdateOwnProfileRequest;
import com.workforce.hrm.dto.request.UserRequestDTO;
import com.workforce.hrm.dto.response.UserResponse;
import org.springframework.web.multipart.MultipartFile;
import java.util.List;
import org.springframework.core.io.Resource;
public interface UserService {

    // =========================================================
    // CREATE USER
    // =========================================================

    UserResponse createUser(
            UserRequestDTO request);


    // =========================================================
    // GET ALL USERS
    // =========================================================

    List<UserResponse> getAllUsers();


    // =========================================================
    // GET USER BY ID
    // =========================================================

    UserResponse getUserById(
            Long id);


    // =========================================================
    // GET CURRENT LOGGED-IN USER
    // =========================================================

    UserResponse getCurrentUser(
            String email);

    UserResponse updateCurrentUser(
            String email,
            UpdateOwnProfileRequest request);


    // =========================================================
    // UPDATE USER
    // =========================================================

    UserResponse updateUser(
            Long id,
            UserRequestDTO request);


    // =========================================================
    // DELETE USER
    // =========================================================

    void deleteUser(
            Long id);


    // =========================================================
    // LOCK USER
    // =========================================================

    void lockUser(
            Long id);


    // =========================================================
    // UNLOCK USER
    // =========================================================

    void unlockUser(
            Long id);


    // =========================================================
    // CHANGE PASSWORD
    // =========================================================

    void changePassword(
            String email,
            ChangePasswordRequest request);


    // =========================================================
    // FORGOT PASSWORD
    // =========================================================

    void forgotPassword(
            String email);


    // =========================================================
    // RESET PASSWORD
    // =========================================================

    void resetPassword(
            ResetPasswordRequest request);
    
 // =========================================================
 // PROFILE PHOTO
 // =========================================================

 String uploadProfilePhoto(
         String email,
         MultipartFile file);

 String getProfilePhoto(
         String email);

 void deleteProfilePhoto(
         String email);

Resource loadProfilePhoto(
        String fileName);
}

File: hrm-portal-frontend/src/services/userService.js
import api from "./api";

// ===============================
// Get All Users
// ===============================
export const getUsers = async () => {
  const response = await api.get("/users");
  return response.data;
};

// ===============================
// Get User By ID
// ===============================
export const getUserById = async (id) => {
  const response = await api.get(`/users/${id}`);
  return response.data;
};

// ===============================
// Create User
// ===============================
export const createUser = async (user) => {
  const response = await api.post("/users", user);
  return response.data;
};

// ===============================
// Update User
// ===============================
export const updateUser = async (id, user) => {
  const response = await api.put(`/users/${id}`, user);
  return response.data;
};

// ===============================
// Delete User
// ===============================
export const deleteUser = async (id) => {
  const response = await api.delete(`/users/${id}`);
  return response.data;
};

// ===============================
// Activate User
// ===============================
export const activateUser = async (id) => {
  const response = await api.put(`/users/${id}/activate`);
  return response.data;
};

// ===============================
// Deactivate User
// ===============================
export const deactivateUser = async (id) => {
  const response = await api.put(`/users/${id}/deactivate`);
  return response.data;
};

// ===============================
// Search Users
// ===============================
export const searchUsers = async (keyword) => {
  const response = await api.get(
    `/users/search?keyword=${keyword}`
  );
  return response.data;
};

// ===============================
// Get Users By Role
// ===============================
export const getUsersByRole = async (role) => {
  const response = await api.get(
    `/users/role/${role}`
  );
  return response.data;
};

// ===============================
// Get Active Users
// ===============================
export const getActiveUsers = async () => {
  const response = await api.get("/users/active");
  return response.data;
};